### PR TITLE
perf(cart): reduce cart item query complexity

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[x] Other... Please describe: perf
```

## What is the current behavior?

Cart item queries are toeing the Magento query complexity limit of 300.

Fixes: #1318 

## What is the new behavior?
Cart item query complexity is now greatly reduced.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: libs/cart/driver/magento/src/models/responses/cart-item.ts [hunk 1]
```diff
@@ -2,14 +2,27 @@ import {
...
+import { MagentoProductStockStatusEnum } from '@daffodil/product/driver/magento';
 
 export enum MagentoCartItemTypeEnum {
 	Simple = 'SimpleCartItem',
 	Bundle = 'BundleCartItem',
 	Configurable = 'ConfigurableCartItem'
 }
 
+export interface MagentoCartItemProduct {
+	__typename: string;
```
Comment (damienwebdev) [inline libs/cart/driver/magento/src/models/responses/cart-item.ts:14]: Can you link where each of these props is used in a transformer? I'm not certain `stock_status` is for example.

**Review by damienwebdev — CHANGES_REQUESTED** [review]

File: libs/cart/driver/magento/src/models/responses/cart-item.ts (hunk 1 above)
Comment (griest024) [reply libs/cart/driver/magento/src/models/responses/cart-item.ts:14]: https://github.com/graycoreio/daffodil/blob/06390525919ab12b3ca77b64401b0f641ba0874f/libs/cart/driver/magento/src/transforms/outputs/cart-item/simple-cart-item-transformer.ts#L32

Comment (griest024) [reply libs/cart/driver/magento/src/models/responses/cart-item.ts:14]: https://github.com/graycoreio/daffodil/blob/06390525919ab12b3ca77b64401b0f641ba0874f/libs/cart/driver/magento/src/transforms/outputs/cart-item/simple-cart-item-transformer.ts#L21

Comment (griest024) [reply libs/cart/driver/magento/src/models/responses/cart-item.ts:14]: https://github.com/graycoreio/daffodil/blob/06390525919ab12b3ca77b64401b0f641ba0874f/libs/cart/driver/magento/src/transforms/outputs/cart-item/simple-cart-item-transformer.ts#L22

Comment (griest024) [reply libs/cart/driver/magento/src/models/responses/cart-item.ts:14]: https://github.com/graycoreio/daffodil/blob/06390525919ab12b3ca77b64401b0f641ba0874f/libs/cart/driver/magento/src/transforms/outputs/cart-item/simple-cart-item-transformer.ts#L26

Comment (griest024) [reply libs/cart/driver/magento/src/models/responses/cart-item.ts:14]: https://github.com/graycoreio/daffodil/blob/06390525919ab12b3ca77b64401b0f641ba0874f/libs/cart/driver/magento/src/transforms/outputs/cart-item/simple-cart-item-transformer.ts#L28

Comment (damienwebdev) [reply libs/cart/driver/magento/src/models/responses/cart-item.ts:14]: Indeed.

**Review by damienwebdev — APPROVED** [review]

**Pull request merged**